agent/handlers/v4/task_metadata_handler.go

```diff
@@ -78,6 +78,22 @@ func TaskMetadataHandler(state dockerstate.TaskEngineState, ecsClient api.ECSCli
 			taskResponse.Containers = responses
 		}
 
+		pulledContainers, ok := state.PulledContainerMapByArn(task.Arn)
+		// Append pulled containers to containers in task response
+		if ok {
+			containerResp := make([]ContainerResponse, 0)
+			for _, dockerContainer := range taskResponse.Containers {
+				containerResp = append(containerResp, dockerContainer)
+			}
+
+			// Convert each pulled container into v4 container response
+			for _, dockerContainer := range pulledContainers {
+				pulledContainerResp := NewPulledContainerResponse(dockerContainer)
+				containerResp = append(containerResp, pulledContainerResp)
```

Comment:
You can append directly to `taskResponse.Containers` here, no need to create a new `containerResp` map